### PR TITLE
changelog: fix formatting

### PR DESCRIPTION
**Pull request by oleg-jukovec**
The patch fixes formatting in the CHANGELOG.md. The key changes:

- Formatting of indents for lists has been unificated.
- A maximum line length is limited to 80 characters except links.
- The release overviews have been rewritten a bit to make them look better.

In addition, several non-critical corrections were made.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,45 +29,51 @@ the way user input is processed.
 
 ## [2.5.2] - 2024-11-07
 
-Update luarocks and libraries version.
+The patch release updates luarocks and libraries version.
```
Comment (patapenka-alexey) [inline CHANGELOG.md:32]:
```suggestion
The release updates luarocks and libraries version.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,45 +29,51 @@ the way user input is processed.
...
 
 ### Fixed
 
 - `tt rocks`: a wrong Lua interpreter is selected.
 
 ## [2.5.1] - 2024-10-31
 
+The patch release updates the MessagePack library in a release build.
```
Comment (patapenka-alexey) [inline CHANGELOG.md:40]:
```suggestion
The release updates the MessagePack library in a release build.
```

**Review by patapenka-alexey — COMMENTED** [review]
Thank you for the patch. Minor typos need to be fixed.

File: CHANGELOG.md (hunk 1 above)
Comment (oleg-jukovec) [reply CHANGELOG.md]: Thank you, updated!

File: CHANGELOG.md (hunk 2 above)
Comment (oleg-jukovec) [reply CHANGELOG.md]: Thank you, updated!

**Review by patapenka-alexey — APPROVED** [review]

**Review by dmyger — APPROVED** [review]

**Pull request merged**